app/controllers/deploys_controller.rb

```diff
@@ -49,6 +49,56 @@ def recent
     end
   end
 
+  # takes the same tokens that are used by the client side filtering
+  # on the recent deploys pages.  returrns a paginated 
+  # json object or CSV of deploys that people are interested in rather than 
+  # doing client side filtering.
+  # params:
+  #   * deployer (name of the user that started the job(s), this is a fuzzy match
+  #   * project_name (name of the project)
+  #   * production (boolean, is this in proudction or not)
+  #   * status (what is the status of this job failed|running| etc)
+
+  def search
+    respond_to do |format|
+
+      # TODO: really need some kinda layer with error handling/messages
+      # etc
+      if (params[:status] && !Job.valid_status?(params[:status]))
+          render json: { :errors => "invalid status given" }, status: 422
+          return
+      end
+      if deployer = params[:deployer] 
+        users = User.where("name LIKE ?", "%#{ActiveRecord::Base.send(:sanitize_sql_like, deployer)}%").pluck(:id)
+      end
+
+      if params[:project_name]
```

Comment:
names exist in the filtering (project and user) which this is intended to emulate.